### DISCUSSION TITLE
Sniff for multiline string concatenation

### DISCUSSION THREAD

**Discussion by jaymcp** [bigbite/phpcs-config#22] [Ideas]
Our string concatenation rules only currently check whether there is an appropriate amount of whitespace around the operator (i.e. 1 space before and after the `.`). We do not currently test beyond that, so the following code is considered valid:
```php
<?php

$string = 'foo'
. "\n" .
'bar';
```

I would argue that this is incorrect; I would expect the above code to be considered valid only if it was formatted as per the one of the examples below:
```php
<?php

$string = 'foo' . "\n" . 'bar';

// or

$string = 'foo'
    . "\n"
    . 'bar';
```

Would it be valuable to folks if we added a Sniff for this style of concatenation, or are we happy to rely upon code review?

**Comment by jrmd** [comment]
I think this one should be implemented, we would likely flag this on code review to be changed so we might as well add a sniff for it.

**Comment by ampersarnie** [comment]
I think this would be a great addition.
